### PR TITLE
feat: Add intelligent workload management policy attributes to advanced_cluster

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Adds two attributes exposing the new Intelligent Workload Management (IWM) fields on `ClusterDescription20240805`: `intelligent_workload_management_policy_overrides` (Optional on `mongodbatlas_advanced_cluster`, Computed on both data sources) and `effective_intelligent_workload_management_policies` (Computed, data sources only).

- Both follow the `searchindexapi.definition.mappings.fields` pattern (`customtypes.MapValue[jsontypes.Normalized]`); each map value is a JSON literal — primitive values are written directly, e.g. `{ LOAD_SHEDDING = true }`.
- Acceptance tests added for LOAD_SHEDDING: `TestAccClusterAdvancedCluster_iwmPolicyOverridesSetOnCreate` and `TestAccClusterAdvancedCluster_iwmPolicyOverridesSetOnUpdate`.
- Both SDK fields (`IntelligentWorkloadManagementPolicyOverrides` and `EffectiveIntelligentWorkloadManagementPolicies`) are typed as `*map[string]any` only in the preview SDK pinned in `go.mod`; they are not yet part of an official SDK release.

Link to any related issue(s): CLOUDP-397755

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fix and verified my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments

### CODE REVIEW

File: .github/workflows/code-health.yml [hunk 1]
```diff
@@ -75,7 +75,7 @@ jobs:
           cat doc.repo.patch
           exit 1
   call-acceptance-tests-workflow:
-    needs: [build, github-linters, unit-test, generate-doc-check]
+    # TEMPORARY DON'T MERGE, needs: [build, github-linters, unit-test, generate-doc-check]
```
[the remark this answers is not in the document]
Comment (lantoli) [reply .github/workflows/code-health.yml:78]: intentional and temporary for this dev-branch PR, noted with a 'TEMPORARY DON'T MERGE' comment, will be reverted before merging to master

File: .github/workflows/acceptance-tests-runner.yml [hunk 2]
```diff
@@ -451,7 +451,8 @@ jobs:
         env:
           MONGODB_ATLAS_LAST_VERSION: ${{ needs.get-provider-version.outputs.provider_version }}
           HTTP_MOCKER_CAPTURE: 'true'
-          ACCTEST_REGEX_RUN: ${{ inputs.reduced_tests == true && '^TestAccMockable' || env.ACCTEST_REGEX_RUN }}
+          # TEMPORARY: don't merge
+          ACCTEST_REGEX_RUN: '^TestAccClusterAdvancedCluster_iwm'
```
[the remark this answers is not in the document]
Comment (lantoli) [reply .github/workflows/acceptance-tests-runner.yml]: intentional and temporary for this dev-branch PR, noted with a 'TEMPORARY: don't merge' comment, will be reverted before merging to master

File: internal/serviceapi/searchindexapi/resource_test.go [hunk 3]
```diff
@@ -66,6 +66,7 @@ func TestAccSearchIndexAPI_withSynonymsUpdatedToEmpty(t *testing.T) {
 	})
 }
 
+// EJECUTAR ESTE
```
[the remark this answers is not in the document]
Comment (lantoli) [reply internal/serviceapi/searchindexapi/resource_test.go]: good catch, unintended change, fixed in 336e171ad

File: internal/serviceapi/searchindexapi/resource_test.go [hunk 4]
```diff
@@ -303,6 +304,8 @@ func configFieldMappingOptionalAnalyzers(projectID, clusterName, indexName strin
             }
         }
     `, projectID, clusterName, indexName, database, collection, analyzers)
+	fmt.Println(ret)
+	panic("stop here")
 }
```
[the remark this answers is not in the document]
Comment (lantoli) [reply internal/serviceapi/searchindexapi/resource_test.go]: good catch, unintended change, fixed in 336e171ad

File: internal/service/advancedcluster/resource_test.go [hunk 5]
```diff
@@ -1398,6 +1452,72 @@ func checksDedicatedNVMeBackupEnabled(projectID, name string, checkPlural bool)
...
 
+func configIWMPolicyOverrides(t *testing.T, projectID, clusterName string, overrides map[string]string) string {
+	t.Helper()
+	var overridesBlock string
+	if overrides != nil {
+		var entries strings.Builder
+		for key, jsonValue := range overrides {
+			fmt.Fprintf(&entries, `
+			%[1]s = jsonencode(%[2]s)`, key, jsonValue)
+		}
```
[the remark this answers is not in the document]
Comment (lantoli) [reply internal/service/advancedcluster/resource_test.go]: fixed in e140f7fa9 (sorted keys via slices.Sorted(maps.Keys(overrides)))

File: internal/service/advancedcluster/model_ClusterDescription20240805.go [hunk 6]
```diff
@@ -54,16 +60,34 @@ func newTFModel(ctx context.Context, input *admin.ClusterDescription20240805, di
 		UseAwsTimeBasedSnapshotCopyForFastInitialSync: types.BoolValue(conversion.SafeValue(input.UseAwsTimeBasedSnapshotCopyForFastInitialSync)),
 		VersionReleaseSystem:                          types.StringValue(conversion.SafeValue(input.VersionReleaseSystem)),
 		PinnedFCV:                                     pinnedFCV,
+		IntelligentWorkloadManagementPolicyOverrides:  iwmOverrides,
 	}
 }
 
+func newIWMPoliciesMap(ctx context.Context, diags *diag.Diagnostics, attrName string, input map[string]any) customtypes.MapValue[jsontypes.Normalized] {
+	if input == nil {
```
[the remark this answers is not in the document]
Comment (lantoli) [reply internal/service/advancedcluster/model_ClusterDescription20240805.go:68]: The Atlas API never returns `{}` for this field when overrides are not set — it either omits the field or returns null. So `newIWMPoliciesMap` treating nil input as Terraform null is correct and there is no risk of perpetual diffs from an empty map in state.

[the remark this answers is not in the document]
Comment (lantoli) [reply internal/service/advancedcluster/model_ClusterDescription20240805.go:68]: This is also covered by the acceptance tests: steps with `checkIWM(nil)` assert `TestCheckNoResourceAttr` on `intelligent_workload_management_policy_overrides.%`, so if Atlas ever returned `{}` instead of null/absent, the check would fail and catch the perpetual diff.

**Review by jromo-mdb — APPROVED** [review]
LGTM

File: docs/resources/advanced_cluster.md [hunk 7]
```diff
@@ -555,6 +555,7 @@ Refer to the following for full privatelink endpoint connection string examples:
...
 * `tags` - (Optional) Set that contains key-value pairs between 1 to 255 characters in length for tagging and categorizing the cluster. See [below](#tags).
+* `intelligent_workload_management_policy_overrides` - (Optional) Map of Intelligent Workload Management (IWM) policy overrides applied on top of Atlas defaults. Each key is an IWM setting and each value is a JSON literal. Primitive values can be written directly; use `jsonencode` only for complex values. Atlas currently supports only `LOAD_SHEDDING` (boolean), which toggles Query Sentinel and Adaptive Operation Rate Limiting, and only on non-sharded clusters using the continuous release system (`cluster_type = "REPLICASET"` and `version_release_system = "CONTINUOUS"`). Atlas defines supported keys, value types, and cluster eligibility rules, and these may change over time. See the [IWM documentation](https://dochub.mongodb.org/core/intelligent-workload-management) for the current list. Example: `intelligent_workload_management_policy_overrides = { LOAD_SHEDDING = true }`.
```
Comment (AgustinBettati) [inline docs/resources/advanced_cluster.md:558]: Should we remove `Atlas currently supports only `LOAD_SHEDDING` (boolean)`? I expect this will age quickly. 
Currently example can be left to showcase use of LOAD_SHEDDING if that is the original motivation.

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: docs/resources/advanced_cluster.md (hunk 7 above)
Comment (AgustinBettati) [inline docs/resources/advanced_cluster.md:558]: q: From [external docs](https://www.mongodb.com/docs/atlas/intelligent-workload-management/) it is not clear that `LOAD_SHEDDING` is a valid key, and its value has to be a boolean. Do we know if this will be the final form of docs, and how users would know the correct structure? This is relevant for both TF and direct API users.

Comment (lantoli) [reply docs/resources/advanced_cluster.md:558]: I added this detail because it's not in the Atlas docs yet, so customers won't be able to use the feature without it. Agreed it will age, but I'd rather have it now and clean it up once the Atlas docs are consolidated. WDYT?

Comment (AgustinBettati) [reply docs/resources/advanced_cluster.md:558]: ack, we can merge to dev branch but would strongly push and track upstream changes to get docs to a clear place. Without docs owned on their side it will be very hard to keep track of improvements and make it usable for TF/API users.

**Comment by lantoli** [comment]
all cluster tests passing: https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/25007403012/job/73233854353
